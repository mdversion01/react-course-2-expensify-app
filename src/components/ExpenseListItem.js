import React from 'react';
import { Link } from 'react-router-dom';
import moment from 'moment';
import numeral from 'numeral';

const ExpenseListItem = ({ id, description, amount, createdAt }) => (
    <div>
        <h3>{description}</h3>
        <p>
            {numeral(amount / 100).format('$0,0.00')}
             - 
             {moment(createdAt).format('MMMM Do, YYYY')}
        </p>

        <Link 
            to={`/edit/${id}`}
        >
            Edit
        </Link>

        
    </div>
);

export default ExpenseListItem;