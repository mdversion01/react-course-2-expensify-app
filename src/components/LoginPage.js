import React from 'react';
import { connect } from 'react-redux';
import { startGoogleLogin, startFbLogin } from '../actions/auth';

export const LoginPage = ({ startGoogleLogin, startFbLogin }) => (
    <div className="box-layout">
        <div className="box-layout__box">
            <h1 className="box-layout__title">Expensify</h1>
            <p>It's time to get your expenses under control.</p>
            <button 
                className="button"
                onClick={startGoogleLogin}
            >
                Login with Google
            </button>

            <button 
                className="button"
                onClick={startFbLogin}
            >
                Login with Facebook
            </button>
        </div>
        
    </div>
);

const mapDispatchToProps = (dispatch) => ({
    startGoogleLogin: () => dispatch(startGoogleLogin()),
    startFbLogin: () => dispatch(startFbLogin())
});

export default connect(undefined, mapDispatchToProps)(LoginPage);