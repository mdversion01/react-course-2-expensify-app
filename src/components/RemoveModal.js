import React from 'react';
import Modal from 'react-modal';

const RemoveModal = (props) => (
    <Modal
        className="modal"
        isOpen={props.showModal}
        contentLabel="Minimal Modal Example"
        onRequestClose={props.handleCloseModal}
        >
        <h3>Remove Expense</h3>
        <p>Do you want to remove this expense?</p>
        <div>
            <button
                className="button modal--button"
                onClick={props.onRemove}
            >
                Remove Expense
            </button>
            <button 
                className="button button--secondary" 
                onClick={props.handleCloseModal}
            >
                Cancel
            </button>
        </div>
    </Modal>
);

export default RemoveModal;