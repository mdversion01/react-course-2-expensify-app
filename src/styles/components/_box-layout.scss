.box-layout {
    align-items: center;
    background: url('/images/bg.jpg');
    background-size: cover;
    display: flex;
    justify-content: center;
    height: 100vh;
    width: 100vw;

    &__box {
        background: fade-out(#FFF, .15);
        border-radius: 3px;
        padding: $lg-size $md-size;
        text-align: center;
        width: 25rem;
    }

    &__title {
        margin: 0 0 $md-size;
        line-height: 1;
    }
}