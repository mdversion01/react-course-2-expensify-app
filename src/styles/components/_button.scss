.button {
    background: $blue;
    border: none;
    border-radius: 2px;
    color: #FFF;
    display: inline-block;
    font-size: $font-size-large;
    font-weight: 300;
    line-height: 1;
    padding: $sm-size;
    text-decoration: none;

    &:hover {
        background: #3179ab;
        text-decoration: none;
    }

    &--link {
        background: none;
    }

    &--secondary {
        background: #888888;

        &:hover {
            background: #616161;
        }
    }

    &--icon {
        background: none;

        &:hover {
            background: #FFF;
        }
    }
}

.icons {
    border: 1px solid $grey;
    border-radius: 3px;
    display: flex;
    justify-content: space-evenly;
    align-items: center;
    padding: 4px;
}
.icon-buttons {
    height: 32px;
    width: 32px;
}