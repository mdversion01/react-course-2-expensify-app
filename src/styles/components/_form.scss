.form {
    display: flex;
    flex-direction: column;
    > * {
        margin-bottom: $md-size;
    }

    &__error {
        margin: 0 0 $sm-size 0;
        font-style: italic;
    }

    &__buttons {
        display: flex;
        justify-content: left;
        align-items: center;
    }
}