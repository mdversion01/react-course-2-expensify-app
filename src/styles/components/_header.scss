.header {
    background: $dark-blue;

    &__title {
        color: #FFF;
        text-decoration: none;

        h1 {
            margin: 0;
        }
    }

    &__content {
        align-items: center;
        display: flex;
        justify-content: space-between;
    }
}