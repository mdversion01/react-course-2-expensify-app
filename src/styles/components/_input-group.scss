.input-group {
    display: flex;
    flex-direction: column;
    margin-bottom: $md-size;
    
    &__item {
       margin-bottom: $sm-size;
       
       @media (min-width: $desktop-breakpoint) {
          margin: 0 $sm-size 0 0;
        }
    }

    @media (min-width: $desktop-breakpoint) {
        flex-direction: row;
    }
}