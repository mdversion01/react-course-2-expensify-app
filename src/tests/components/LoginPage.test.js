import React from 'react';
import { shallow } from 'enzyme';
import { LoginPage } from '../../components/LoginPage';

test('Should correctly render login page', () => {
    const wrapper = shallow(<LoginPage />);
    expect(wrapper).toMatchSnapshot();
});

test('Should correctly call startGoogleLogin on button click', () => {
    const startGoogleLogin = jest.fn();
    const wrapper = shallow(<LoginPage startGoogleLogin={startGoogleLogin} />);
    wrapper.find('.google').simulate('click');
    expect(startGoogleLogin).toHaveBeenCalled();
});

test('Should correctly call startFbLogin on button click', () => {
    const startFbLogin = jest.fn();
    const wrapper = shallow(<LoginPage startFbLogin={startFbLogin} />);
    wrapper.find('.fb').simulate('click');
    expect(startFbLogin).toHaveBeenCalled();
});