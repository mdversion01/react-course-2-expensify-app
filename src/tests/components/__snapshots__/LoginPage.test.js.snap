// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should correctly render login page 1`] = `
<div
  className="box-layout"
>
  <div
    className="box-layout__box"
  >
    <h1
      className="box-layout__title"
    >
      Expensify
    </h1>
    <p>
      It's time to get your expenses under control.
    </p>
    <div>
      Login with:
    </div>
    <div
      className="icons"
    >
      <button
        className="google button button--icon"
      >
        <img
          className="icon-buttons"
          src="/images/google-icon.png"
          title="Login with Google"
        />
      </button>
      <button
        className="fb button button--icon"
      >
        <img
          className="icon-buttons"
          src="/images/fb-icon.png"
          title="Login with Facebook"
        />
      </button>
    </div>
  </div>
</div>
`;
